datafusion/core/tests/sql/joins.rs

```diff
@@ -1254,32 +1373,31 @@ async fn join_with_hash_unsupported_data_type() -> Result<()> {
     );
 
     let expected = vec![
-        "+----+-----+-----+------------+----+-----+-----+------------+",
-        "| c1 | c2  | c3  | c4         | c1 | c2  | c3  | c4         |",
-        "+----+-----+-----+------------+----+-----+-----+------------+",
-        "| 1  | aaa | 100 | 1970-01-02 | 1  | aaa | 100 | 1970-01-02 |",
-        "| 2  | bbb | 200 | 1970-01-03 | 2  | bbb | 200 | 1970-01-03 |",
-        "| 3  | ccc | 300 | 1970-01-04 | 3  | ccc | 300 | 1970-01-04 |",
-        "+----+-----+-----+------------+----+-----+-----+------------+",
+        "+------------+------------+---------+-----+------------+------------+------------+-----------+---------+------------+",
+        "| c1         | c2         | c3      | c4  | c5         | c1         | c2         | c3        | c4      | c5         |",
+        "+------------+------------+---------+-----+------------+------------+------------+-----------+---------+------------+",
+        "|            |            |         |     |            |            |            | 100000.00 | abcdefg |            |",
+        "|            |            |         |     |            |            | 1970-01-04 | 0.00      | qwerty  | 74776f     |",
+        "|            | 1970-01-04 | 789.00  | ghi |            | 1970-01-04 |            | 789.00    |         | 7468726565 |",
+        "| 1970-01-04 |            | -123.12 | jkl | 7468726565 | 1970-01-02 | 1970-01-02 | -123.12   | abc     | 6f6e65     |",
+        "+------------+------------+---------+-----+------------+------------+------------+-----------+---------+------------+",
     ];
 
     let results = execute_to_batches(&ctx, sql).await;
     assert_batches_sorted_eq!(expected, &results);
```

Comment:
This test is failing on CI: https://github.com/apache/arrow-datafusion/runs/6953453815?check_suite_focus=true

Looking at the diff it appears to be related to column c5: 
![Screen Shot 2022-06-19 at 7 45 59 AM](https://user-images.githubusercontent.com/490673/174479269-31385d43-a730-4a21-a10e-ccbfbf42b0a0.png)

And thus it seems unrelated to this PR (though a real bug). Thus  I personally suggest you change the query to not select `c5` and we then file ticket to chase down the real problem.

Reply:
I agree. That wrong c5 is not actually a correct row in t1.

>and we then file ticket to chase down the real problem.

@AssHero cannot reproduce this might also be a clue. And I hope we can add c5 back in the next fixing PR.

Reply:
> https://github.com/apache/arrow-datafusion/runs/6953453815?check_suite_focus=tr
Thanks! Let me fix the test case.